File: docs/snippets/java/cli/CliWithConfig.java
```java
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public final class CliWithConfig {
  private static final ObjectMapper MAPPER = new ObjectMapper();

  private CliWithConfig() {}

  public static JsonNode extractWithConfig(String filePath, String configPath)
      throws IOException, InterruptedException {
    ProcessBuilder pb = new ProcessBuilder(
        "kreuzberg",
        "extract",
        filePath,
        "--config",
        configPath,
        "--format",
        "json");
    pb.redirectErrorStream(true);

    Process process = pb.start();

    StringBuilder output = new StringBuilder();
    try (BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()))) {
      String line;
      while ((line = reader.readLine()) != null) {
        output.append(line);
      }
    }

    int exitCode = process.waitFor();
    if (exitCode != 0) {
      throw new RuntimeException("CLI exited with code " + exitCode + ": " + output);
    }

    return MAPPER.readTree(output.toString());
  }

  public static void main(String[] args) throws IOException, InterruptedException {
    String configFile = "kreuzberg.toml";
    String document = "document.pdf";

    System.out.println("Extracting " + document + " with config " + configFile);
    JsonNode result = extractWithConfig(document, configFile);

    String content = result.get("content").asText();
    System.out.println("Content length: " + content.length());
    System.out.println("Format: " + result.get("format").asText());
    System.out.println("Languages: " + result.get("languages").toString());
  }
}
```

File: docs/snippets/python/cli/cli_with_config.py
```python
import subprocess
import json
import sys
from pathlib import Path

def extract_with_config(file_path: str, config_path: str) -> dict:
    result = subprocess.run(
        ["kreuzberg", "extract", file_path, "--config", config_path, "--format", "json"],
        capture_output=True,
        text=True,
    )

    if result.returncode != 0:
        print(f"Error: {result.stderr}", file=sys.stderr)
        sys.exit(1)

    return json.loads(result.stdout)

config_file = Path("kreuzberg.toml")
document = "document.pdf"

print(f"Extracting {document} with config {config_file}")
result = extract_with_config(str(document), str(config_file))

print(f"Content length: {len(result.get('content', ''))}")
print(f"Format: {result.get('format')}")
print(f"Languages: {result.get('languages')}")
```
